internal/action/run_trigger.go

```diff
@@ -0,0 +1,167 @@
+			}
+
+			trigger.SourceID = fmt.Sprintf("${data.tfe_workspace.%s.id}", rt.SourceName)
+		}
+	} else {
+		return nil, fmt.Errorf("run trigger source ID or source name must be set")
+	}
+
+	return trigger, nil
+}
+
+// MergeRunTriggers returns a list of run triggers objects, one per workspace per trigger
+func MergeRunTriggers(inputs RunTriggerInputs, workspaceInputs map[string]RunTriggerInputs, workspaces []*Workspace, organization string) (RunTriggers, error) {
+	triggers := RunTriggers{}
+
+	for _, input := range inputs {
+		for _, ws := range workspaces {
+			rt, err := input.ToRunTrigger(ws, workspaces, organization)
+			if err != nil {
+				return nil, err
+			}
+
+			triggers = append(triggers, *rt)
+		}
+	}
+
+	for wsName, wsInputs := range workspaceInputs {
+		ws := FindWorkspace(workspaces, wsName)
+		if ws == nil {
+			return nil, fmt.Errorf("failed to match workspace variable with known workspaces. Workspace %s not found", ws.Workspace)
+		}
+
+		for _, wsi := range wsInputs {
+			rt, err := wsi.ToRunTrigger(ws, workspaces, organization)
+			if err != nil {
+				return nil, err
+			}
+
+			triggers = append(triggers, *rt)
+		}
+	}
+
+	return triggers, nil
+}
+
+// ToResource returns a tfeprovider.RunTrigger object from the calling RunTrigger object
+func (t RunTrigger) ToResource() *tfeprovider.RunTrigger {
+	return &tfeprovider.RunTrigger{
+		WorkspaceID:  fmt.Sprintf("${tfe_workspace.workspace[%q].id}", t.Workspace.Workspace),
+		SourceableID: t.SourceID,
+	}
+}
+
+// AppendRunTriggers takes a list of run triggers and adds them to the passed module
+func AppendRunTriggers(module *tfconfig.Module, triggers RunTriggers) {
+	if len(triggers) == 0 {
+		return
+	}
+
+	triggerForEach := map[string]tfeprovider.RunTrigger{}
+
+	for _, t := range triggers {
+		if t.DataRef != nil {
+			for name, ref := range t.DataRef {
+				module.AppendData("tfe_workspace", name, ref)
+			}
+		}
+
+		triggerForEach[fmt.Sprintf("%s-%s", t.Workspace.Workspace, t.SourceID)] = *t.ToResource()
+	}
+
+	rt := tfeprovider.RunTrigger{
+		ForEach:      triggerForEach,
+		SourceableID: "${each.value.sourceable_id}",
+		WorkspaceID:  "${each.value.workspace_id}",
+	}
+
+	module.AppendResource("tfe_run_trigger", "trigger", rt)
+}
+
+// FetchInboundRunTriggers takes a workspace and returns related tfe.RunTrigger objects
+func FetchInboundRunTriggers(ctx context.Context, client *tfe.Client, workspaceID string) ([]*tfe.RunTrigger, error) {
+	rts, err := client.RunTriggers.List(ctx, workspaceID, tfe.RunTriggerListOptions{
+		ListOptions: tfe.ListOptions{
+			PageSize: maxPageSize,
+		},
+		RunTriggerType: tfe.String("inbound"),
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return rts.Items, nil
+}
+
+// FindRelatedRunTriggers returns a list of inbound run triggers associated with the passed workspace
+func FindRelatedRunTriggers(ctx context.Context, client *tfe.Client, workspace *Workspace, organization string) (RunTriggers, error) {
+	triggers := RunTriggers{}
```

Comment:
```suggestion
func FindRelatedRunTriggers(ctx context.Context, client *tfe.Client, workspace *Workspace, organization string) (triggers RunTriggers, err error) {
```

Reply:
If you need an empty return value, just name it and you can refer to it in the function

Reply:
whaaat thats so cool